src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs

```diff
@@ -46,14 +46,15 @@ public void Dispose()
 
       public virtual void Wait(CancellationToken token)
```

Comment:
Hopefully a little refactoring will show what the intention is here.
We are not relying on the cancellationToken to throw an exception at all now. We use the token to get a waithandle. We get a second waithandle from the process handle. Together these waithandles will tell us when the task has finished and what was the cause.